src/app/[slug]/page.tsx

```diff
@@ -0,0 +1,175 @@
+    { label: 'My Articles' },
+    {
+      label: 'Favorited Articles',
+    },
+  ] as const;
+
+  const fetchFavoritedData = async (slug: string) => {
+    try {
+      const response = await getFavorited(slug);
+      if (!response.ok) {
+        throw new Error('Failed to fetch profile');
+      }
+      return response.json();
+    } catch (error) {
+      console.error(error);
+      throw error;
+    }
+  };
+
+  useEffect(() => {
+    const fetchFavorited = async () => {
+      try {
+        const data: ProfileResponseType =
+          await fetchFavoritedData(pathOnlyUserName);
+        setFavoritedData(data);
+        // 여기서 data 변수에 접근 가능
+      } catch (error) {
+        console.error(error);
+      }
+    };
+    if (currentTab === 'Favorited Articles') {
+      fetchFavorited();
+    }
+  }, [currentTab]);
+
+  const [authorData, setAuthorData] = useState<undefined | FeedResponseType>();
+
+  const fetchAuthorData = async (author: string) => {
+    try {
+      const response = await getAuthorArticle(author);
+      if (!response.ok) {
+        throw new Error('Failed to fetch profile');
+      }
+      return response.json();
+    } catch (error) {
+      console.error(error);
+      throw error;
+    }
+  };
+
+  useEffect(() => {
+    const fetchAuthor = async () => {
+      try {
+        const data: FeedResponseType = await fetchAuthorData(pathOnlyUserName);
+        setAuthorData(data);
+        // 여기서 data 변수에 접근 가능
+      } catch (error) {
+        console.error(error);
+      }
+    };
+    if (currentTab === 'My Articles') {
+      fetchAuthor();
+    }
+    fetchAuthor();
+  }, [currentTab]);
+
+  const getCurrentTabData = (currentTab: TapType) => {
+    if (currentTab === 'My Articles') {
+      return authorData;
+    }
+    return favoritedData;
+  };
+
+  const getDataAndSetData = (
+    setState: Dispatch<SetStateAction<FeedResponseType>>,
+  ) => {
+    return function (data: FeedResponseType) {
+      console.log('나한테 왔어욤');
+      if (data) {
+        setState(data);
+      }
+    };
+  };
+
+  console.log(authorData);
+
+  return (
+    <div className="profile-page">
+      {<UserProfile />}
+      <div className="container">
+        <div className="row">
+          <div className="col-xs-12 col-md-10 offset-md-1">
+            <div className="articles-toggle">
+              <ul className="nav nav-pills outline-active">
+                {PROFILE_PAGE_TAB.map((tab) => (
+                  <li
+                    key={tab.label}
+                    onClick={() => setCurrentTab(tab.label)}
+                    className="nav-item"
+                  >
+                    <p
+                      className={cn(
+                        CurrentTabStyle({
+                          isCurrentTab: currentTab === tab.label,
+                        }),
+                      )}
+                    >
+                      {tab.label}
+                    </p>
+                  </li>
+                ))}
```

Comment:
가능하면 JSX를 깔끔하게 작성해주면 좋을 것 같아요.

```suggestion
                {PROFILE_PAGE_TAB.map(({ label }) => (
                  <li
                    key={label}
                    onClick={() => setCurrentTab(label)}
                    className="nav-item"
                  >
                    <p
                      className={cn(
                        CurrentTabStyle({
                          isCurrentTab: currentTab === label,
                        }),
                      )}
                    >
                      {label}
                    </p>
                  </li>
                ))}
```